src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java

```diff
@@ -362,6 +378,51 @@ public static ThumbnailGenerationTask getBitmapWorkerTask(ImageView imageView) {
         return null;
     }
 
+    public static Bitmap addVideoOverlay(Bitmap thumbnail){
+        Bitmap playButton = BitmapFactory.decodeResource(MainApp.getAppContext().getResources(),
+                R.drawable.view_play);
+
+        Bitmap resizedPlayButton = Bitmap.createScaledBitmap(playButton,
```

Comment:
> Thumbnails are saved always with the same resolution in a given device, right? In that case, the scaled load of the 'play' image and the calculation for its coordinates should be done only once, and kept in static fields.

As video thumbnails can occur in file *and* grid view, the computation must be in this function, as it depends on the thumbnail size.